snapcraft/internal/elf.py

```diff
@@ -283,25 +289,37 @@ def wrapper(*args, **kwargs):
 class Patcher:
     """Patcher holds the necessary logic to patch elf files."""
 
-    def __init__(self, *, dynamic_linker: str, root_path: str) -> None:
+    def __init__(self, *, dynamic_linker: str, root_path: str,
+                 preferred_patchelf_path=None) -> None:
         """Create a Patcher instance.
 
         :param str dynamic_linker: the path to the dynamic linker to set the
                                    elf file to.
         :param str root_path: the base path for the snap to determine
                               if use of $ORIGIN is possible.
+        :param str preferred_patchelf_path: patch the necessary elf_files with
+                                        this patchelf.
         """
         self._dynamic_linker = dynamic_linker
         self._root_path = root_path
 
-        # If we are running from the snap we want to use the patchelf
+        # We will first fallback to the preferred_patchelf_path,
+        # if that is not found we will look for the snap and finally,
+        # if we are running from the snap we want to use the patchelf
         # bundled there as it would have the capability of working
         # anywhere given the fixed ld it would have.
         # If not found, resort to whatever is on the system brought
         # in by packaging dependencies.
         # The docker conditional will work if the docker image has the
         # snaps unpacked in the corresponding locations.
-        if common.is_snap():
+        if preferred_patchelf_path:
+            self._patchelf_cmd = preferred_patchelf_path
+        # We use the full path here as the path may not be set on
+        # build systems where the path is recently created and added
+        # to the environment
+        elif os.path.exists('/snap/bin/pathelf'):
```

Comment:
I doubt you'll ever find the `pathelf` command :wink: .

Speaking of that, I could imagine it being helpful  in bug reports to, once we've decided which patchelf we're using, provide a debug print saying "using patchelf: {patchelf_path}".